subsys/storage/stream/stream_flash.c

```diff
@@ -137,18 +151,18 @@ static int flash_sync(struct stream_flash_ctx *ctx)
 		/* Invert to ensure that caller is able to discover a faulty
 		 * flash_read() even if no error code is returned.
 		 */
-		for (int i = 0; i < ctx->buf_bytes; i++) {
+		for (int i = 0; i < buf_bytes_aligned; i++) {
 			ctx->buf[i] = ~ctx->buf[i];
 		}
 
 		rc = flash_read(ctx->fdev, write_addr, ctx->buf,
-				ctx->buf_bytes);
+				buf_bytes_aligned);
 		if (rc != 0) {
 			LOG_ERR("flash read failed: %d", rc);
 			return rc;
 		}
 
-		rc = ctx->callback(ctx->buf, ctx->buf_bytes, write_addr);
+		rc = ctx->callback(ctx->buf, buf_bytes_aligned, write_addr);
```

Comment:
I have doubts regarding passing aligned value to the callback: shouldn't that be unaligned value of ctx->buf_bytes?
buf_bytes represents number of bytes that has been passed to write by user, so this is count of bytes user know, the alignment may add additional bytes to check by callback that the user may not be able to verify, for example:
```
uint8_t buf[] = { 1, 3 };
```
if user decides to do flash 2 byte write with flush, or flush afterwards, the flash will be written, on 4 byte aligned flash, as { 1, 3, erase_value, erase_value }, now the user will be asked, via callback, to verify something else than has been requested because the end of buffer is different.

Reply:
I agree, it makes sense that it should be `buf_bytes` instead, so I will update it. Should I update it for the  `flash_read()` above too?
I think this is an issue with the existing code as well, since it uses the incremented `buf_bytes`. So I can add a test to verify it.